### ISSUE TITLE
False negative - `unwrap_used` doesn't trigger on macros that return Result

### ISSUE THREAD

**Issue by junbl** [rust-lang/rust-clippy#13455]
### Summary

Clippy does not report a use of unwrap if it's called on the result of a macro (at least declarative macros).

Not 100% sure, but I believe this is a regression since I remember having `write!().unwrap()` calls that used to lint.

Looks like `expect_used` has the same issue, too.

### Lint Name

unwrap_used
expect_used

### Reproducer

I tried this code:

[playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5058493768bdf90aea78f3f7e368c3de)

```rust

macro_rules! fallible {
    () => {
        Ok::<_, ()>(())
    };
}

fn fallible() -> Result<(), ()> {
    Ok(())
}

#[deny(clippy::unwrap_used)]
fn main() {
    fallible!().unwrap();
    
    fallible().unwrap();
}

```

I expected to see this happen: Both the calls to `fallible!()` and `fallible()` would lint.

Instead, this happened: Only `fallible()` lints, no lint for the macro. `#[expect]` reports "unfulfilled lint."

### Version

rustc 1.81.0 (eeb90cda1 2024-09-04)
binary: rustc
commit-hash: eeb90cda1969383f56a2637cbd3037bdf598841c
commit-date: 2024-09-04
host: x86_64-unknown-linux-gnu
release: 1.81.0
LLVM version: 18.1.7
